tools/partutil/extend_partition.go

```diff
@@ -0,0 +1,136 @@
+package partutil
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+//ExtendPartition extends a partition to a specific end sector
+func ExtendPartition(disk string, partNumInt, end int) error {
+	if len(disk) <= 0 || partNumInt <= 0 {
+		return errors.New("invalid disk name or partition number")
+	}
+
+	//for cases like loop5p1
+	partNum := strconv.Itoa(partNumInt)
+	if disk[len(disk)-1] >= '0' && disk[len(disk)-1] <= '9' {
+		partNum = "p" + partNum
+	}
+
+	partName := disk + partNum
+
+	//dump partition table to a file
+	cmd := string("sudo sfdisk --dump ") + disk + " > extend_partition_tmp"
+	err := ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+	defer os.Remove("extend_partition_tmp")
+
+	err = editPartitionTableFile("extend_partition_tmp", partName, end)
+	if Check(err, "editing partition table file of "+partName+" to ending sector at: "+strconv.Itoa(end)) {
+		return err
+	}
+
+	//write partition table back
+	cmd = "sudo sfdisk " + disk + " < " + " extend_partition_tmp "
+	err = ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+	log.Printf("\nCompleted extending %s\n\n", partName)
+
+	// check and repair file system in the partition
+	cmd = "sudo e2fsck -fp " + partName
+	err = ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+	log.Printf("\nCompleted checking file system of %s\n\n", partName)
+
+	// resize file system in the partition
+	cmd = "sudo resize2fs " + partName
+	err = ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+
+	log.Printf("\nCompleted updating file system of %s\n\n", partName)
+	return nil
+}
+
+//change partition table file to extend partition
+func editPartitionTableFile(fileName, partName string, end int) error {
+	in, err := ioutil.ReadFile(fileName)
+	if Check(err, "cannot read partition table file") {
+		return err
+	}
+
+	lines := strings.Split(string(in), "\n")
+	have := false //whether has valid information about the partition
+	for i, line := range lines {
```

Comment:
It would be helpful to give an example of a partition_file so that it will be better to corelate with the code.

Reply:
Something like:

/dev/sda8 : start=       86016, size=       32768, type=0FC63DAF-8483-4772-8E79-3D69D8477DE4, uuid=6ADF6DFA-2060-9F44-A405-E33AFB28E0AE, name="OEM"